main.py

```diff
@@ -75,23 +78,38 @@ def perform_query_1(s3_client, s3_file_list):
 
 ### Returns a list of records from a given file
 def extract_records_from_file(s3_client, filename):
-    s3_file = s3_client.get_object(
-        Bucket=S3_BUCKET,
-        Key=filename,
-    )
-    return list(s3_file['Body'].iter_lines()) ### iter_lines() is significantly faster than read().splitlines()
+    if USE_LOCAL_DATA:
+        with open(filename, 'r') as f:
+            return f.readlines()
+    else:
+        s3_file = s3_client.get_object(
+            Bucket=S3_BUCKET,
+            Key=filename,
+        )
+        return list(s3_file['Body'].iter_lines()) ### iter_lines() is significantly faster than read().splitlines()
 
 ### Returns filenames from an S3 list files (list_objects) query
 def list_s3_files_matching_prefix(s3_client, prefix_string):
-    response = list_s3_objects(s3_client, prefix_string)
-    filenames = []
-    if response.get('Contents'):
-        [filenames.append(item['Key']) for item in response.get('Contents')]
-    while response.get('NextContinuationToken'):
-        response = list_s3_objects(s3_client, prefix_string, response.get('NextContinuationToken'))
+    if USE_LOCAL_DATA:
+        try:
+            files_and_directories = glob.glob(LOCAL_DATA_REPOSITORY+"/"+prefix_string+"/**/*", recursive=True)
+            files_only = []
+            for filepath in files_and_directories:
+                if os.path.isfile(filepath):
```

Comment:
This seems to return file only in the current directory. Don't we need an `else` part to recursively scan the directories?

Reply:
`glob.glob()` returns a list of everything that satisfies the constraints given, which in this case returns both directories and files. This if statement check makes sure that only files are returned, not directories.

Reply:
So does it return files inside those directories too? If so, then the logic is correct. If not, we will be skipping those files without a recursive `else` part.

Reply:
Yes the glob.glob recursively returns all files and folders inside all subdirectories as well. It returns the same file count as actual S3 query, so I believe the logic to be correct.